web_poet/testing/pytest.py

```diff
@@ -0,0 +1,120 @@
+
+    @staticmethod
+    def sorted(items: List["WebPoetItem"]) -> List["WebPoetItem"]:
+        """Sort the test list by the test name."""
+        return sorted(items, key=operator.attrgetter("name"))
+
+    def collect(self) -> Iterable[Union[pytest.Item, pytest.Collector]]:  # noqa: D102
+        result: List[WebPoetItem] = []
+        for entry in self.path.iterdir():
+            if entry.is_dir():
+                item: WebPoetItem = WebPoetItem.from_parent(
+                    self,
+                    name=entry.name,
+                    path=entry,
+                )
+                if item.input_path.is_dir() and item.output_path.is_file():
+                    result.append(item)
+        return self.sorted(result)
+
+
+class WebPoetItem(pytest.Item):
+    """Represents a directory containing one test."""
+
+    @property
+    def type_name(self) -> str:
+        """The name of the type being tested."""
+        if not self.parent:
+            raise ValueError("WebPoetItem has no parent")
+        return self.parent.name
+
+    @property
+    def input_path(self) -> Path:
+        """The inputs subdirectory path."""
+        return self.path / INPUT_DIR_NAME
+
+    @property
+    def output_path(self) -> Path:
+        """The output file path."""
+        return self.path / OUTPUT_FILE_NAME
+
+    @property
+    def meta_path(self) -> Path:
+        """The metadata file path."""
+        return self.path / META_FILE_NAME
+
+    def get_page(self) -> ItemPage:
+        """Return the page object created from the saved input."""
+        cls = load_class(self.type_name)
+        if not issubclass(cls, ItemPage):
+            raise TypeError(f"{self.type_name} is not a descendant of ItemPage")
+        storage = SerializedDataFileStorage(self.input_path)
+        return deserialize(cls, storage.read())
+
+    def get_meta(self) -> dict:
+        """Return the test metadata."""
+        if not self.meta_path.exists():
+            return {}
+        return json.loads(self.meta_path.read_bytes())
+
+    def get_output(self) -> dict:
+        """Return the output from the recreated Page Object."""
+        po = self.get_page()
+        item = asyncio.run(ensure_awaitable(po.to_item()))
+        return ItemAdapter(item).asdict()
+
+    def get_expected_output(self) -> dict:
+        """Return the saved output."""
+        return json.loads(self.output_path.read_bytes())
+
+    def runtest(self) -> None:  # noqa: D102
+        meta = self.get_meta()
+        frozen_time: Optional[str] = meta.get("frozen_time")
+        if frozen_time:
+            with freeze_time(frozen_time):
+                output = self.get_output()
+        else:
+            output = self.get_output()
+        expected_output = self.get_expected_output()
+        assert output == expected_output
+
+
+_found_po_dirs: Set[Path] = set()
+
+
+def pytest_collect_file(
+    file_path: Path, path: Any, parent: pytest.Collector
```

Comment:
Yeah, the `path` argument is deprecated, not sure if I should support passing it. I guess only pytest itself calls this method and with the 7.0+ version it will pass the `file_path` argument, but we probably need to support older pytest too? Not sure about this.

Reply:
According to https://github.com/pytest-dev/pytest/discussions/10261 it's not enough to handle both arguments, old pytest requires the old signature.